src/clients/FlexConfigurationClient.ts

```diff
@@ -1,31 +1,103 @@
+import phin from 'phin';
 import { ConfigurationContext, ConfigurationInstance } from 'twilio/lib/rest/flexApi/v1/configuration';
 
+import { TwilioCliError } from '../exceptions';
+
+interface Auth {
+  accountSid: string;
+  username: string;
+  password: string;
+}
+
 /**
  * Wrapper Twilio Flex Configuration Public API
  */
 export default class FlexConfigurationClient {
   private client;
 
-  constructor(client: ConfigurationContext) {
+  private auth: Auth;
+
+  constructor(client: ConfigurationContext, auth: Auth) {
     this.client = client;
+    this.auth = auth;
   }
 
   /**
    * Fetches the {@link ConfigurationInstance}
    */
   public async fetch(): Promise<ConfigurationInstance> {
-    return this.client.fetch();
+    const config = await this.client.fetch();
+    if (!config.serverlessServiceSids) {
+      config.serverlessServiceSids = [];
+    }
+
+    return config;
   }
 
   /**
    * Fetches the Serverless ServiceSid
    */
   public async getServerlessSid(): Promise<string | null> {
     const config = await this.fetch();
-    if (!config.serverlessServiceSids) {
-      return null;
-    }
 
     return config.serverlessServiceSids[0];
   }
+
+  /**
+   * Registers Serverless sid
+   * @param serviceSid the sid to register
+   */
+  public async registerServerlessSid(serviceSid: string): Promise<ConfigurationInstance> {
+    const config = await this.fetch();
+    if (config.serverlessServiceSids.includes(serviceSid)) {
+      return config;
+    }
+
+    config.serverlessServiceSids.push(serviceSid);
+    await this.updateServerlessSids(config.serverlessServiceSids);
+
+    return this.fetch();
+  }
+
+  /**
+   * Removes a Serverless sid
+   * @param serviceSid the sid to remove
+   */
+  public async unregisterServerlessSid(serviceSid: string): Promise<ConfigurationInstance> {
+    const config = await this.fetch();
+    const index = config.serverlessServiceSids.indexOf(serviceSid);
```

Comment:
```
if (!config.serverlessServiceSids.includes(serviceSid)) {
  return config;
}
```

Reply:
I do need the index if it does exist to remove; hence why I'm using this.

Reply:
oops didn't notice you were using it below 👍